flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/SinkUtil.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.plan.util
+
+import java.util.{List => JList}
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment
+import org.apache.flink.streaming.api.transformations.{SinkTransformation, StreamTransformation}
+
+import scala.collection.JavaConverters._
+
+/**
+  * Util for [[org.apache.flink.table.plan.nodes.calcite.Sink]]s.
+  */
+object SinkUtil {
+
+  /**
+    * Gets the newly added sink transformation. The returned sink transformation will be used
+    * to set resources.
+    */
+  def getNewlyAddedSinkTransformation(env: StreamExecutionEnvironment): StreamTransformation[_] = {
```

Comment:
It looks very hack.

Reply:
Yes. I also left a comment on the code. Under the currently design, we have to return a result transformation for every node (including Sink, mainly used to set resources). I have discussed with @godfreyhe offline and came up with this idea.

If we refactor the node design (fallback to previous design: exclude sink to return a transformation), it will introduce a lot of changes and may delay the progress.

Reply:
Looks we need a better solution